### DISCUSSION TITLE
Triggering auto update permission dialog

### DISCUSSION THREAD

**Discussion by Cornfritter** [sparkle-project/Sparkle#2343] [Q&A] [answered]
Hi- Hoping someone can help.

I'm working with a sandboxed app on Monterey, and the auto update permission dialog is not being presented on 2nd run as the documentation says.

Are there known issues which might prevent it from being presented?

How do you reset to trigger the auto update permission dialog subsequently? I've tried trashing the app preferences plist or app container folder with no success.

I've followed the documentation carefully, and everything else seems to work -- checking for updates manually, etc.

Thanks a lot!

**Comment by zorgiepoo** [comment, marked as answer]
There aren't any known issues.

> How do you reset to trigger the auto update permission dialog subsequently? I've tried trashing the app preferences plist or app container folder with no success.

You can't modify the files on disk as they may persist in memory in some system daemons. You need to manipulate defaults using the `defaults` command.

Check your app's defaults with `defaults read my-app-bundle-id`

Specifically for this question you are interested in

```
defaults read my-app-bundle-id SUEnableAutomaticChecks
defaults read my-app-bundle-id SUHasLaunchedBefore
```

You can use `defaults delete my-app-bundle-id SUEnableAutomaticChecks` e.g. to remove that key from the defaults.

The permission dialog won't show up if you set `SUEnableAutomaticChecks` in your app's info.plist (this will override it) or if you force an early call to [automaticallyChecksForUpdates](https://sparkle-project.org/documentation/api-reference/Classes/SPUUpdater.html#/c:objc(cs)SPUUpdater(py)automaticallyChecksForUpdates) or force an early update check.

**Reply by Cornfritter** [reply]
Thanks so much! Will try this.

**Reply by Cornfritter** [reply]
Quick followup: you mentioned setting SUEnableAutomaticChecks in info.plist will override the permission dialog. I just want to confirm that this works both when setting it true and false? I read in an earlier post that SUEnableAutomaticChecks only works when set false, to disable auto checks, and that users still need to grant permission to enable them. Maybe that was old behavior. Thanks again!

**Reply by zorgiepoo** [reply]
Do not put SUEnableAutomaticChecks in your Info.plist at all if you want Sparkle to ask users for permission to check for updates automatically.

If the key is present in your Info.plist users will not be asked for permission to check for updates, and instead automatic update checking will be set to what value you specify there. That is what I mean by override.

**Reply by Cornfritter** [reply]
Got it. Just wanted to check how it works. Many thanks!
